x-pack/elastic-agent/pkg/core/plugin/process/start.go

```diff
@@ -134,6 +136,15 @@ func (a *Application) start(ctx context.Context, t app.Taggable, cfg map[string]
 	if err != nil {
 		return err
 	}
+
+	// Hook to JobObject on windows, noop on other platforms.
+	// This ties the application processes lifespan to the agent's.
+	// Fixes the orphaned beats processes left behind situation
+	// after the agent process gets killed.
+	if err := proc.JobObject.Assign(a.state.ProcessInfo.Process); err != nil {
```

Comment:
Any way this can be moved to `x-pack/elastic-agent/pkg/core/process`.

When executing a cmd on linux `Pdeathsig: syscall.SIGKILL` is set, giving the same result as you are getting on windows. Would be great to have the uniformity at the same level.

Reply:
absolutely, will move

Reply:
actually need this code in osquerybeat as well, would not want to put this under ```x-pack/elastic-agent```
maybe something like ```x-pack/common``` if we have it

Reply:
I was talking about just these lines, it could have been integrated in a better way removing this specific call in the process start flow. It could just have been a standard way of how Elastic Agent starts subprocesses on Windows.

This means that this is only used for running of subprocess applications. Things like running the endpoint-security installer or the nested elastic-agent for enrollment and fleet-server bootstrap will not use this feature.

Reply:
@blakerouse ahh, got it. 
I didn't move that to `x-pack/elastic-agent/pkg/core/process` because was not 100% sure of other possible uses at that time. Sure, can move and open another PR.

Reply:
> [x-pack/libbeat](https://github.com/elastic/beats/tree/main/x-pack/libbeat) would be the appropriate place.

@ph I will move from libbeat to x-pack/libbeat in the other PR then.